controllers/workspace/status.go

```diff
@@ -138,3 +140,20 @@ func getIdeUrl(exposedEndpoints map[string]v1alpha1.ExposedEndpointList) string
 	}
 	return ""
 }
+
+func getInfoMessage(workspace *devworkspace.DevWorkspace, conditions map[devworkspace.WorkspaceConditionType]string) string {
+	var failedStartMsg string
+	for conditionType, conditionMessage := range conditions {
+		switch conditionType {
+		// Take error condition message as overriding failed start message
+		case devworkspace.WorkspaceError:
+			return conditionMessage
+		case devworkspace.WorkspaceFailedStart:
+			failedStartMsg = conditionMessage
+		}
+	}
+	if failedStartMsg != "" {
+		return failedStartMsg
+	}
+	return workspace.Status.IdeUrl
```

Comment:
Should we return IdeURL when it's not ready yet?

Reply:
That's what we currently do -- personally I like it because I can copy it and refresh a few times to load a workspace, rather than repeatedly using `kubectl get`, but I'm happy to do it either way (and with your suggestion above of reporting latest condition, it makes sense to not put ideURL until it's started anyways)

Reply:
Probably it's another issue, but I saw we return outdated URL, after routing class is changed but workspace is stopped.

and with proposal to report the latest condition, I just wondered how we can avoid confusion we got  when workspace is started but not started, because Che Routing is not available yet.

There we may also report finalizing a workspace, currently the way is to check removal timestamp )